src/Hng.Web/Controllers/UserController.cs

```diff
@@ -47,5 +47,26 @@ public async Task<ActionResult<UserDto>> CreateUser([FromBody] UserCreationDto b
             var response = await _mediator.Send(command);
             return CreatedAtAction(nameof(CreateUser), response);
         }
+
+        [HttpPost("login")]
+        [ProducesResponseType(typeof(UserLoginResponseDto), StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(UserLoginResponseDto), StatusCodes.Status400BadRequest)]
+        public async Task<ActionResult<UserLoginResponseDto>> Login([FromBody] UserLoginRequestDto loginRequest)
+        {
+            var command = new CreateUserLoginCommand(loginRequest);
+            var response = await _mediator.Send(command);
+
+            if (response.Data == null)
+            {
+                return BadRequest(new
+                {
+                    message = "Invalid credentials",
+                    error = "Invalid email or password.",
+                    status_code = StatusCodes.Status400BadRequest
```

Comment:
The response here should not be `Bad request` it should be `401` Unauthorized

Reply:
Also this should not be in user controller. You route will now be `/api/v1/users/login`. Is this how it's supposed to be?

Reply:
ok sir buh i dont understand very well sir